wallet/src/de/schildbach/wallet/service/BlockchainServiceImpl.java

```diff
@@ -1481,4 +1490,103 @@ private void closeStream(InputStream mnlistinfoBootStrapStream) {
             }
         }
     }
+
+    // TODO: should we have a backup blockchain file?
+//    private NewBestBlockListener newBestBlockListener = block -> {
+//        try {
+//            backupBlockStore.put(block);
+//        } catch (BlockStoreException x) {
+//            throw new RuntimeException(x);
+//        }
+//    };
+
+    private boolean verifyBlockStore(BlockStore store) throws BlockStoreException {
+            StoredBlock cursor = store.getChainHead();
+            for (int i = 0; i < 10; ++i) {
+                cursor = cursor.getPrev(store);
+                if (cursor == null || cursor.getHeader().equals(Constants.NETWORK_PARAMETERS.getGenesisBlock())) {
+                    break;
+                }
+            }
+            return true;
+    }
+
+    private boolean verifyBlockStore(BlockStore store, ScheduledExecutorService scheduledExecutorService) {
+        try {
+            ScheduledFuture<Boolean> future = scheduledExecutorService.schedule(() -> verifyBlockStore(store), 100, TimeUnit.MILLISECONDS);
+            return future.get(1, TimeUnit.SECONDS);
+        } catch (Exception e) {
+            log.warn("verification of blockstore failed:", e);
+            return false;
+        }
+    }
```

Comment:
This will verify that the blockStore's can be accessed and give a time limit.  When the blockchain was accessed during sync, the thread froze.  So this was added to check for the freeze earlier.

Turns out, though, that with one user that this check seemed to "warm up" the blockStore and the sync thread didn't freeze later.